Comparison/src/cz/fidentis/comparison/localAreas/LocalAreaLibrary.java

```diff
@@ -0,0 +1,386 @@
+                                                       (float )triangle[0].getY(),
+                                                       (float )triangle[0].getZ())));
+                i = -(n.dot(p0) + d) / n.dot(pp);
+            } else {
+                return null;
+            }
+            
+            if (i >= 0) {
+                // intersection with plane
+                Vector3f intP = new Vector3f( p0.x + i * pp.x, 
+                        p0.y + i * pp.y,
+                        p0.z + i * pp.z);
+
+                //PI-T0
+                Vector3f w = new Vector3f(intP.getX() - (float )triangle[0].getX(), 
+                    intP.getY() - (float )triangle[0].getY(),
+                    intP.getZ() - (float )triangle[0].getZ());
+
+                float uv = u.dot(v);
+                float uu = u.dot(u);
+                float vv = v.dot(v);
+                float wu = w.dot(u);
+                float wv =  w.dot(v);
+                float s = (uv * wv - vv * wu) / (uv * uv - uu * vv);
+                float t = (uv * wu - uu * wv) / (uv * uv - uu * vv);
+
+                if (s >= 0 && t >= 0 && s + t <= 1) {
+                      return intP;
+                }
+            }
+        }
+        return null;
+    }
+    
+    
+    /**
+     * Slight modification of 2D algorithm
+     * @param verticesIndexes indexes
+     * @param model model
+     * @return 
+     */
+    public static List<Point3D> giftWrapping(List<Integer> verticesIndexes, Model model){
+        
+        //getting points
+        List<Point3D> points = new ArrayList();
+        for (int i = 0; i < verticesIndexes.size(); i++) {
+           Point3D point = new Point3D( model.getVerts().get(verticesIndexes.get(i)).x, 
+                                        model.getVerts().get(verticesIndexes.get(i)).y, 
+                                        model.getVerts().get(verticesIndexes.get(i)).z);
+           points.add(point);
+        }
+        
+        List<Point3D> vertexList = new ArrayList<>();
+        
+        int pivot = smallestX(points);
+        int currentIndex = pivot;
+        int previousIndex = pivot;
+        
+        Point3D a = points.get(currentIndex);
+        vertexList.add(a);
+
+        List<Integer> indexes = new ArrayList<>();
+        
+        //going around all points to make a wrapping
+        do {
+            int nextIndex = getMinimalAnglePoint(points, currentIndex, previousIndex, indexes);
+            indexes.add(nextIndex);
+            if (currentIndex != previousIndex){
+                a = points.get(currentIndex);
+                vertexList.add(a);
+                
+            }
+            previousIndex = currentIndex;
+            currentIndex = nextIndex;
+            
+        } while (pivot != currentIndex);
+        
+        
+        return vertexList;
+    }
+    
+    private static Vector3f crossProduct(Vector3f u, Vector3f v){
```

Comment:
tato metoda sa uz nachadza v module Utils, cz.fidentis.utils.MathUtils, pouzij metodu z Util, aby sa odstranili duplikaty

Reply:
tuto prehodim do utils, nieje modifikovana ako niektore ostatne

Reply:
giftWrapping odstranit zavistlost na Model (miesto toho List<Vector3f>) a hodit to do cz.fidentis.utils.MeshUtils v module Utils

Reply:
done